### ISSUE TITLE
Explicitly included packages with `--package`  don't run with the `--workspace` flag when they are `publish = false`.

### ISSUE THREAD

**Issue by suaviloquence** [obi1kenobi/cargo-semver-checks#868]
### Steps to reproduce the bug with the above code

In pwd [`test_crates/manifest_tests/workspace_all_publish_false`](https://github.com/obi1kenobi/cargo-semver-checks/tree/main/test_crates/manifest_tests/workspace_all_publish_false):

```sh
cargo semver-checks --manifest-path new --baseline-root old --workspace --package a --verbose
```

Package `a` (marked as `publish = false` in Cargo.toml) is explicitly included with the `--package a` flag.

### Actual Behaviour

`a` is not semver-checked:

```
    Skipping a v0.1.0 (current)
    Skipping b v0.1.0 (current)
```

### Expected Behaviour

`a` is semver-checked, similar to running the command without `--workspace`:

```
     Parsing a v0.1.0 (current)
      Parsed [   0.584s] (current)
     Parsing a v0.1.0 (baseline)
      Parsed [   0.558s] (baseline)
    Checking a v0.1.0 -> v0.1.0 (no change)
    Starting 80 checks, 0 unnecessary on 16 threads
        PASS [   0.001s]       major        auto_trait_impl_removed
--- snip ---
        PASS [   0.001s]       major        variant_marked_non_exhaustive
     Checked [   0.013s] 80 checks: 79 pass, 1 fail, 0 warn, 0 skip

--- failure function_missing: pub fn removed or renamed ---

Description:
A publicly-visible function cannot be imported by its prior path. A `pub use` may have been removed, or the function itself may have been renamed or removed entirely.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#item-remove
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.34.0/src/lints/function_missing.ron

Failed in:
  function a::should_not_run, previously in file /home/m/dev/cargo-semver-checks/test_crates/manifest_tests/workspace_all_publish_false/old/a/src/lib.rs:4

     Summary semver requires new major version: 1 major and 0 minor checks failed
    Finished [   1.170s] a
```

with a 
```
  

### Generated System Information

#### Software version

cargo-semver-checks 0.34.0 (a0c0a37)

#### Operating system

Linux 6.10.3-zen1-2-zen

#### Command-line

```bash
/home/m/dev/cargo-semver-checks/target/debug/cargo-semver-checks semver-checks --bugreport
```

#### cargo version

```
> cargo -V
cargo 1.80.0 (376290515 2024-07-16)
```

#### Compile time information

- Profile: debug
- Target triple: x86_64-unknown-linux-gnu
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-unknown-linux-gnu

**Comment by suaviloquence** [comment]
https://github.com/obi1kenobi/cargo-semver-checks/blob/74635ee19dc135130d08f932d19f3da0ec951b1d/src/main.rs#L418-L425

The `--package` and `--workspace` flags are being handled as exclusive here, where `--workspace` takes precedence.  Do we want to change this behavior?

**Comment by obi1kenobi** [comment]
Just to make sure, let's check with @epage. Context for you, Ed: some of our users expressed confusion that:
- If using `--workspace`, the `--package` flag is silently ignored.
- If they have a `publish = false` package in a workspace, attempting to scan it with `--workspace --package X` still results in the package being skipped. Normally, `--package X` means the package gets scanned even if it's `publish = false`.

This behavior seems suboptimal no matter how we interpret the meaning of the flags.

What behavior would you recommend so that we stay as close as possible to cargo? Should we error out if a `--workspace --package` combo is used, or obey `--package` and scan the package despite its `publish = false`?

**Comment by suaviloquence** [comment]
There is also the `--all` flag which is currently equivalent to `--workspace`

**Comment by epage** [comment]
For cargo's behavior, see https://github.com/rust-lang/cargo/blob/9e152bba503c9c2e195d32b7e751ac723d7d1d7e/src/cargo/ops/cargo_compile/packages.rs#L34
